demo/src/features/debug3d/jobs/prepare.rs

```diff
@@ -1,51 +1,55 @@
-use super::write::Debug3dCommandWriter;
-use crate::features::debug3d::{
-    Debug3dDrawCall, Debug3dRenderFeature, Debug3dUniformBufferObject, Debug3dVertex,
-    ExtractedDebug3dData,
-};
+use rafx::render_feature_prepare_job_predule::*;
+
+use super::{Debug3DVertex, Debug3DWriteJob, LineList3D, RenderFeatureType};
 use crate::phases::OpaqueRenderPhase;
 use rafx::api::{RafxBufferDef, RafxMemoryUsage, RafxResourceType};
 use rafx::framework::{MaterialPassResource, ResourceArc};
-use rafx::nodes::{
-    FeatureCommandWriter, FeatureSubmitNodes, FramePacket, PrepareJob, RenderFeature,
-    RenderFeatureIndex, RenderJobPrepareContext, RenderView, ViewSubmitNodes,
-};
 
-pub struct Debug3dPrepareJobImpl {
+pub type Debug3dUniformBufferObject = shaders::debug_vert::PerFrameUboUniform;
+
+pub struct Debug3DPrepareJob {
     debug3d_material_pass: ResourceArc<MaterialPassResource>,
-    extracted_debug3d_data: ExtractedDebug3dData,
+    line_lists: Vec<LineList3D>,
 }
 
-impl Debug3dPrepareJobImpl {
+impl Debug3DPrepareJob {
     pub(super) fn new(
         debug3d_material_pass: ResourceArc<MaterialPassResource>,
-        extracted_debug3d_data: ExtractedDebug3dData,
+        line_lists: Vec<LineList3D>,
     ) -> Self {
-        Debug3dPrepareJobImpl {
+        Debug3DPrepareJob {
             debug3d_material_pass,
-            extracted_debug3d_data,
+            line_lists,
         }
     }
 }
 
-impl<'a> PrepareJob for Debug3dPrepareJobImpl {
+impl<'a> PrepareJob for Debug3DPrepareJob {
     fn prepare(
         self: Box<Self>,
         prepare_context: &RenderJobPrepareContext,
         _frame_packet: &FramePacket,
         views: &[RenderView],
-    ) -> (Box<dyn FeatureCommandWriter>, FeatureSubmitNodes) {
-        profiling::scope!("Debug3d Prepare");
+    ) -> (Box<dyn WriteJob>, FeatureSubmitNodes) {
+        profiling::scope!(super::PREPARE_SCOPE_NAME);
+
+        let mut writer = Box::new(Debug3DWriteJob::new(
+            self.debug3d_material_pass.clone(),
+            self.line_lists.len(),
+        ));
 
         let mut descriptor_set_allocator = prepare_context
             .resource_context
             .create_descriptor_set_allocator();
+
         let per_view_descriptor_set_layout =
             &self.debug3d_material_pass.get_raw().descriptor_set_layouts
                 [shaders::debug_vert::PER_FRAME_DATA_DESCRIPTOR_SET_INDEX];
 
-        let mut per_view_descriptor_sets = Vec::default();
-        for view in views {
+        for view in views
+            .iter()
+            .filter(|view| view.feature_is_relevant::<RenderFeatureType>())
```

Comment:
I kind of dislike having this typedef. It makes it impossible to search the code for all occurrences of a particular feature. I do realize this is something that has to be changed when copy-pasting an existing feature to make a new one, but the person doing it would already need to search-replace OldFeatureName with NewFeatureName and old_feature_name with new_feature_name so I don't see it as a big deal.

Reply:
(this comment is meant for all usages of RenderFeatureType as a typedef, not just this particular example)

Reply:
Really? I felt like it made it _easier_ to search the code.

With the typedef, when you search for a feature, you'll find exactly the following:
1. The definition of it inside of the feature `mod.rs` file.
2. Any external uses of it by importing or reference.

Without the typedef, when you search for a feature, you'll find the following:
1. The definition of it inside of the feature `mod.rs` file.
2. Any external uses of it by importing or reference.
3. An arbitrary number of internal uses of the feature type inside of the feature directory, drowning out everything else.

e.g. compare `MeshRenderFeature` 

![image](https://user-images.githubusercontent.com/4888530/114279757-15616200-99eb-11eb-928b-e634298fb445.png)

vs

![image](https://user-images.githubusercontent.com/4888530/114279845-74bf7200-99eb-11eb-9b78-af7b779c2559.png)

Reply:
Yeah, the typedef makes it so searching for MeshRenderFeature omits the internal usages which is misleading. If it’s used in 15 places, I do want to see all those 15 places. :)

Reply:
:( Very well, I'll nuke the typedef later today.